aodncore/pipeline/storage.py

```diff
@@ -101,6 +112,51 @@ def set_is_overwrite(self, pipeline_files, dest_path_attr='dest_path'):
             abs_path = self._get_absolute_dest_path(pipeline_file=pipeline_file, dest_path_attr=dest_path_attr)
             pipeline_file.is_overwrite = self._get_is_overwrite(pipeline_file, abs_path)
 
+    def download(self, remote_pipeline_files, local_path):
+        """Download the given RemotePipelineFileCollection or RemotePipelineFile from the storage backend
+
+        :param remote_pipeline_files: collection to download
+        :param local_path: local path to download the files into
+        :return: None
+        """
+        download_collection = ensure_remotepipelinefilecollection(remote_pipeline_files)
+
+        self._pre_run_hook()
+
+        for remote_pipeline_file in download_collection:
+            try:
+                self._download_file(remote_pipeline_file=remote_pipeline_file, local_path=local_path)
+                remote_pipeline_file.local_path = os.path.join(local_path, self.prefix, remote_pipeline_file.dest_path)
```

Comment:
I don't think this line is quite right: the `self.prefix` trumps the `local_path` before it.
Actually `LocalFileStorageBroker._download_file()` already sets the `local_path` attribute, so at least for that case, you don't need to do it here too.
You also don't have this line in `download_iterator()` below (line 150).